src/backends/pytorch/libtorch_backend.cc

```diff
@@ -0,0 +1,745 @@
+                                         "' to Torch datatype");
+  }
+
+  // torch::TensorOptions options =
+  //     torch::TensorOptions().device(device_).dtype(pr.second);
+  torch::Tensor input_tensor = torch::from_blob(content, shape, pr.second);
+  input_tensor = input_tensor.to(device_);
+
+  if (input_tensor.nbytes() != byte_size) {
+    return Status(
+        RequestStatusCode::INTERNAL,
+        "unexpected size " + std::to_string(byte_size) +
+            " for inference input '" + name + "', expecting " +
+            std::to_string(input_tensor.nbytes()));
+  }
+  inputs_.push_back(input_tensor);
+
+  return Status::Success;
+}
+
+Status
+LibTorchBackend::Context::ReadFixedSizedOutputTensor(
+    const std::string& name, const int& op_index, const DataType dtype, const size_t dtype_byte_size,
+    const size_t total_batch_size, std::vector<Scheduler::Payload>* payloads)
+{
+  std::vector<int64_t> content_shape;
+  char* content = nullptr;
+  size_t byte_size = 0;
+  RETURN_IF_ERROR(
+      GetOutputTensor(name, op_index, dtype, &content, &byte_size, &content_shape));
+
+  const size_t total_byte_size =
+      GetElementCount(content_shape) * dtype_byte_size;
+  const size_t batch1_byte_size = total_byte_size / total_batch_size;
+
+  if (byte_size != total_byte_size) {
+    return Status(
+        RequestStatusCode::INTERNAL,
+        "unexpected size for output '" + name + "', byte-size " +
+            std::to_string(byte_size) + " does not equal " +
+            std::to_string(total_batch_size) + " * " +
+            std::to_string(batch1_byte_size));
+  }
+
+  size_t content_offset = 0;
+
+  for (auto& payload : *payloads) {
+    const InferRequestHeader& request_header =
+        payload.request_provider_->RequestHeader();
+    const size_t expected_byte_size =
+        request_header.batch_size() * batch1_byte_size;
+
+    // If 'payload' requested this output then copy it from
+    // 'content'. If it did not request this output then just
+    // skip it in the 'content'.
+    if ((payload.response_provider_ != nullptr) &&
+        payload.response_provider_->RequiresOutput(name)) {
+      void* buffer;
+      Status status = payload.response_provider_->AllocateOutputBuffer(
+          name, &buffer, expected_byte_size, content_shape);
+      if (status.IsOk()) {
+        memcpy(buffer, content + content_offset, expected_byte_size);
+      }
+
+      if (!status.IsOk()) {
+        payload.status_ = status;
+      }
+    }
+
+    content_offset += expected_byte_size;
+  }
+
+  return Status::Success;
+}
+
+Status
+LibTorchBackend::Context::GetOutputTensor(
+    const std::string& name, const int& op_index, const DataType dtype, char** content,
+    size_t* byte_size, std::vector<int64_t>* content_shape)
+{
+  torch::DeviceType output_device = torch::kCPU;
+  // TODO: Fix for supporting multiple outputs
+  try {
+    outputs_[op_index] = outputs_[op_index].to(output_device);
```

Comment:
Will document this. In case of multiple models, the user has to ensure that the model config file contains the outputs in the same order as in the model.